src/index.js

```diff
@@ -1,33 +1,242 @@
+  saveTaskstoLocalStorage();
+  if (checkbox.checked) {
+    listItem.querySelector('.text').classList.add('completed');
+  } else {
+    listItem.querySelector('.text').classList.remove('completed');
+  }
+  saveTaskstoLocalStorage();
+};
+
+activitySection.addEventListener('change', handleCheckboxchange);
+
+const loadActivityArrayFromLocalStorage = () => {
+  const activityArrayString = localStorage.getItem('activities');
+  if (activityArrayString) {
+    return JSON.parse(activityArrayString);
+  }
+  return [];
+};
+
+const activityArray1 = loadActivityArrayFromLocalStorage();
+
+activityArray1.forEach((task, index) => {
+  const listItem = document.getElementById(index);
+  if (task.completed) {
+    listItem.querySelector('.text').classList.add('completed');
+  }
+});
+
+// deleteing a task
+
+const DeleteTask = (e) => {
+  if (e.target.classList.contains('trash')) {
+    const listItem = e.target.parentNode.parentNode;
+    const taskIndex = listItem.getAttribute('id');
+    // const taskIndex = Array.from(listItem.parentNode.children).indexOf(listItem) - 1;
+    activityArray.splice(taskIndex, 1);
+    reIndex();
+    saveTaskstoLocalStorage();
+    AddToScreen();
+  }
+};
+activitySection.addEventListener('click', DeleteTask);
+// deleting all checked tasks
+
+const deleteCheckedTasks = () => {
+  activityArray = activityArray.filter((MyConstructor) => !MyConstructor.completed);
+  saveTaskstoLocalStorage();
+  AddToScreen();
+};
+// button for dleting all checked tasks
+const cLearALL = () => {
+  deleteCheckedTasks();
+  reIndex();
+  saveTaskstoLocalStorage();
+  AddToScreen();
+};
+clearAll.addEventListener('click', cLearALL);
+// refresh function
+// const reFreshPage = (e) => {
+//   if (e.target.classList.contains('refresh')) {
+//     reFresh.setAttribute('animation', 'spin');
+//     window.location.reload();
+//     window.addEventListener('load', () => {
+//       reFresh.setTimeout(() => reFresh.classList.remove('animation'), 4500);
+//     });
+//   }
+// };
+document.addEventListener('click', reFreshPage);
+
+// editing text in task
+const EditTask = (e) => {
+  const parent = e.target.parentNode.parentNode;
+  const index = parent.getAttribute('id');
+  if (e.key === 'Enter' && e.target.classList.contains('input-edit')) {
+    const newDescription = e.target.value;
+    // editTask(newDescription, index);
+    activityArray[index].description = newDescription;
+    saveTaskstoLocalStorage();
+    parent.innerHTML = `
+    <span class="activity td">
+    <input type="checkbox" name="" class="check-box"
+    ${activityArray[index].completed ? 'checked' : ''}>
+    <span class="text">${activityArray[index].description}</span>
+    <i class="fa-solid fa-ellipsis-vertical"></i></span>
+`;
+    parent.classList.add('listitem');
+    parent.classList.remove('edit');
+  }
+};
+activitySection.addEventListener('keypress', EditTask);
+// MyConstructor();
+// AddtoLocalStorage();
+// saveTaskstoLocalStorage(); // AddToScreen();
```

Comment:
- [ ] Please kindly remove all commented code from your codebase as it does not follow best practices and corrupts the codebase.